src/v/debug_bundle/debug_bundle_service.cc

```diff
@@ -109,6 +110,41 @@ bool is_valid_k8s_namespace(std::string_view ns) {
     return !ns.empty() && ns.size() <= max_ns_length && is_valid_rfc1123(ns);
 }
 
+bool is_valid_k8s_label_name_or_value(std::string_view name) {
+    constexpr auto max_label_name_size = 63;
+    constexpr std::string_view label_name_pattern
+      = R"(^([a-zA-Z0-9]([a-zA-Z0-9-_.]{0,61}[a-zA-Z0-9])?))";
+
+    return !name.empty() && name.size() <= max_label_name_size
+           && RE2::FullMatch(name, RE2(label_name_pattern));
+}
+
+// This function doesn't fully check that the label prefix is a 100% valid K8S
+// label prefix but at least verifies that the contained characters are valid.
+// rpk will validate using the golang labels.go library
+bool is_likely_valid_k8s_label_prefix(std::string_view prefix) {
+    constexpr auto prefix_max_length = 253;
+    constexpr std::string_view label_name_pattern
+      = R"(^([a-zA-Z0-9]([a-zA-Z0-9-_.]{0,251}[a-zA-Z0-9])?))";
+    return !prefix.empty() && prefix.size() <= prefix_max_length
+           && RE2::FullMatch(prefix, RE2(label_name_pattern));
```

Comment:
Could the bodies of these two function be merged into a single one?

They seem to be parameterize-able on the length of the max_length. I assume that the benefit of having constexpr values is negligible compared to the runtime of regex.

Reply:
If the refactoring is worth doing, potentially merge and and/alias `is_valid_rfc1123` into `is_valid_k8s_namespace`, and make the `RE2`s `static const` so that they are compiled once.